dataactbroker/permissions.py

```diff
@@ -146,6 +148,52 @@ def wrapped(submission_id, *args, **kwargs):
     return inner
 
 
+def requires_agency_perms(perm):
+    """ Decorator that checks the current user's permissions and validates them against the agency code. It expects an
+         existing_submission_id, cgac_code, or frec_code parameter on top of the function arguments.
+
+        Args:
+            perm: the type of permission we are checking for
+
+        Returns:
+            The args/kwargs that were initially provided
+
+        Raises:
+            ResponseException: If the user doesn't have permission to access the submission at the level requested
+                or no valid agency code was provided.
+    """
+    def inner(fn):
+        @requires_login
+        @wraps(fn)
+        def wrapped(*args, **kwargs):
+            sess = GlobalDB.db().session
+            req_args = webargs_parser.parse({
+                'existing_submission_id': webargs_fields.Int(missing=None),
+                'cgac_code': webargs_fields.String(missing=None),
+                'frec_code': webargs_fields.String(missing=None)
+            })
+
```

Comment:
Would it make sense to check here if all three are null to start? If so, we probably want to return a different error message, like "Must provide either a valid cgac/frec code or an existing submission ID." Or is that too much detail?

Reply:
If you think it'd be more clear, I can do that. But it wouldn't affect the speed or anything because nothing really happens between line 174 and 187.

Reply:
If you think this is clear enough about what the problem is then this is fine, I was just wondering if we should cover that base or not.